Source/WebCore/Modules/mediastream/MediaStreamTrack.cpp

```diff
@@ -367,15 +398,20 @@ static MediaConstraints createMediaConstraints(const std::optional<MediaTrackCon
 
 void MediaStreamTrack::applyConstraints(const std::optional<MediaTrackConstraints>& constraints, DOMPromiseDeferred<void>&& promise)
 {
-    auto completionHandler = [this, protectedThis = Ref { *this }, constraints, promise = WTFMove(promise)](auto&& error) mutable {
+    if (m_ended) {
+        promise.reject(Exception { InvalidAccessError, "Track has ended"_s });
+        return;
+    }
+
+    m_private->applyConstraints(createMediaConstraints(constraints), [protectedThis = Ref { *this }, constraints, promise = WTFMove(promise)](auto&& error) mutable {
```

Comment:
This is a preexisting issue but we should probably queue a task here like we are doing in ImageCapture::takePhoto.